theano/tensor/slinalg.py

```diff
@@ -79,6 +97,13 @@ def grad(self, inputs, gradients):
         dz = gradients[0]
         chol_x = self(x)
 
+        # Replace the cholesky decomposition with 1 if there are nans
+        # or solve_upper_triangular will throw a ValueError.
+        if self.on_error == 'nan':
+            ok = tensor.all(tensor.nlinalg.diag(chol_x) > 0)
```

Comment:
ok = tensor.isnan(tensor.nlinalg.diag(cho_x))

why why check only the diag? the perform set the full output to nan.